### DISCUSSION TITLE
Java but using jbang instead of mvn/gradle

### DISCUSSION THREAD

**Discussion by maxandersen** [pulumi/pulumi#9562] [Preview Features]
I’m trying to enable use of jbang.dev as a way to run pulumi scripts to avoid requiring several files and directory levels when a single file can suffice.

I found https://github.com/pulumi/pulumi-java/blob/main/pkg/cmd/pulumi-language-java/main.go And can update it to have understanding of jbang, but I’m wondering how I test this best ? 

What’s the easiest to “install” my own pulumi-language-java version into a pulumi install ?

**Comment by maxandersen** [comment]
And alternative would be if there existed a way to run any arbitrary command, is there such “language” plugin ?

**Comment by t0yv0** [comment]
If you have your own pulumi-language-java in PATH it will use that, which is a nice way to experiment.

We might be interested in merging support into https://github.com/pulumi/pulumi-java/blob/main/pkg/cmd/pulumi-language-java/main.go if you get it working.

**Reply by maxandersen** [reply]
https://github.com/pulumi/pulumi-java/pull/575 has what I could get working for now.
